src/pages/actions/useAllowReload.tsx

```diff
@@ -0,0 +1,10 @@
+import { useEffect } from "react";
+
+export default function useAllowReload(
+  condition: boolean,
+  setAllowReload: (arg: boolean) => void
+) {
+  useEffect(() => {
+    condition ? setAllowReload(false) : setAllowReload(true);
```

Comment:
> Why is this not just: setAllowReload(!condition)?

Oh my, so true!

> Why extract it in a custom hook if it is that simply? :D

I think I started by noticing that I needed the same hook in two places. Maybe it is so simple that it doesn't merit a custom hook, but this way we can't possibly modify one and forget the other one.

Reply:
> Maybe it is so simple that it doesn't merit a custom hook, but this way we can't possibly modify one and forget the other one.

That makes sense. I would suggest then making the custom hook aware of the way it is used, i.e. passing all variables in that are relevant (the dialogs?) instead of a generic "condition". Currently, the boolean logic is duplicated in both places.